### PR TITLE
pythonPackages.digital-ocean: 1.13.2 -> 1.15.0

### PR DESCRIPTION
**Pull request by Kiwi**
updated to newest version (1.13.2 was released ~2.25 years ago). switched to fetchFromGitHub to enable tests. added kiwi to maintainers. formatted with nixpkgs-fmt

###### Motivation for this change
it was old. and i think i need it updated for work i want to do on https://github.com/Kiwi/nixops-digitalocean/tree/digitalocean

###### Things done
i tested it (the python2Packages one at least) with my nixops-digitalocean successfully and the tests are now enabled and pass for both python2Packages and python3Packages. doing a nixpkgs-review pr 81217 right now... it might take a while.

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/python-modules/digitalocean/default.nix
Comment (jonringer) [inline pkgs/development/python-modules/digitalocean/default.nix]: for expressions in `python-modules`, please import the packages individually
```suggestion
, jsonpickle
, requests
, pytest
, responses
```

File: pkgs/development/python-modules/digitalocean/default.nix
Comment (jonringer) [inline pkgs/development/python-modules/digitalocean/default.nix]:
```suggestion
    pytest
```

File: pkgs/development/python-modules/digitalocean/default.nix
Comment (jonringer) [inline pkgs/development/python-modules/digitalocean/default.nix]: I think it's fine here to have it in one line. But I don't feel strongly about it
```suggestion
    maintainers = with maintainers; [ kiwi teh ];
```

File: pkgs/development/python-modules/digitalocean/default.nix
Comment (jonringer) [inline pkgs/development/python-modules/digitalocean/default.nix]:
```suggestion
    homepage = "https://pypi.python.org/pypi/python-digitalocean";
```

File: pkgs/development/python-modules/digitalocean/default.nix
Comment (Kiwi) [reply pkgs/development/python-modules/digitalocean/default.nix]: I'm in the habit and a fan of lists being on own line I blame @worldofpeace

File: pkgs/development/python-modules/digitalocean/default.nix
Comment (Kiwi) [reply pkgs/development/python-modules/digitalocean/default.nix]: I, too, am a fan of quoted urls and usually change it. I missed this one... thank you

File: pkgs/development/python-modules/digitalocean/default.nix
Comment (Kiwi) [reply pkgs/development/python-modules/digitalocean/default.nix]: How come we do this btw? Easier to see what it depends on?

File: pkgs/development/python-modules/digitalocean/default.nix
Comment (jonringer) [reply pkgs/development/python-modules/digitalocean/default.nix]: I don't feel strongly about this, you're a maintainer, I'll defer to your preference :)

File: pkgs/development/python-modules/digitalocean/default.nix
Comment (jonringer) [reply pkgs/development/python-modules/digitalocean/default.nix]: There's a lot of cases in applications where you want to change the version of a single version. Makes it easier to override individual dependencies.

**Review by jonringer — APPROVED** [review]
diff LGTM
commit LGTM
```
[7 built, 0.0 MiB DL]
https://github.com/NixOS/nixpkgs/pull/81217
6 package built:
nixops nixopsUnstable nixops_1_6_1 python27Packages.digital-ocean python37Packages.digital-ocean python38Packages.digital-ocean
```

**Comment by jonringer** [comment]
@GrahamcOfBorg build nixops nixopsUnstable nixops_1_6_1 python27Packages.digital-ocean python37Packages.digital-ocean python38Packages.digital-ocean

File: pkgs/development/python-modules/digitalocean/default.nix
Comment (worldofpeace) [inline pkgs/development/python-modules/digitalocean/default.nix]:
```suggestion
  preCheck = ''
    cd digitalocean
  '';
```

File: pkgs/development/python-modules/digitalocean/default.nix
Comment (worldofpeace) [inline pkgs/development/python-modules/digitalocean/default.nix]:
```suggestion
  checkInputs = [
    pytest
    responses
    pytestCheckHook
  ] ++ stdenv.lib.optionals (!isPy3k) [
    mock
  ];
```

just runs pytest nicely.

**Review by worldofpeace — CHANGES_REQUESTED** [review]

File: pkgs/development/python-modules/digitalocean/default.nix
Comment (Kiwi) [reply pkgs/development/python-modules/digitalocean/default.nix]: i tried this and it didn't work

Comment (jonringer) [reply pkgs/development/python-modules/digitalocean/default.nix]: that's because the `setuptoolsCheckPhase` is still being ran, you would either need to do `dontUseSetuptoolsCheck = true;` or, instead of changing directory, you could almost move/delete the digitalocean directory.

Comment (Kiwi) [reply pkgs/development/python-modules/digitalocean/default.nix]: oh ok thanks. i updated it now

File: pkgs/development/python-modules/digitalocean/default.nix
Comment (worldofpeace) [inline pkgs/development/python-modules/digitalocean/default.nix]: you shouldn't set the derivation here?
```suggestion
```

**Comment by worldofpeace** [comment]
This PR should be good after that final comment is applied.

File: pkgs/development/python-modules/digitalocean/default.nix
Comment (Kiwi) [reply pkgs/development/python-modules/digitalocean/default.nix]: @cleverca22 said it makes debugging easier

Comment (worldofpeace) [reply pkgs/development/python-modules/digitalocean/default.nix]: I believe it's very intentional that the default name is `source` in the fetchers.
See https://github.com/NixOS/nixpkgs/commit/c3255fe8ec326d2c8fe9462d49ed83aa64d3e68f. If it makes debugging easier, you can just change it when debugging.

Comment (jonringer) [reply pkgs/development/python-modules/digitalocean/default.nix]: I think what they meant was, it's impossible to have a bad fixed-point derivation. Having the name as "source" can create an instance where you change the version of the out package, but the fetchurl is still retrieving the old outputs because there's a cached instance. Passing an explicit name will create a new .drv and there will be a hash-mismatch if you forget to then update the sha256

Comment (jonringer) [reply pkgs/development/python-modules/digitalocean/default.nix]: I believe https://github.com/NixOS/nixpkgs/commit/c3255fe8ec326d2c8fe9462d49ed83aa64d3e68f was an attempt to standardize the fetcher derivation names, not make it more ergonomic

Comment (Kiwi) [reply pkgs/development/python-modules/digitalocean/default.nix]: yeah what jon said :+1:

Comment (worldofpeace) [reply pkgs/development/python-modules/digitalocean/default.nix]: If we feel so strongly about changing this for fetchers derivations, this PR isn't the place to spread this. All it does is confuse contributors and divide our codebase. If we think https://github.com/NixOS/nixpkgs/commit/c3255fe8ec326d2c8fe9462d49ed83aa64d3e68f wasn't a good choice, let's change that everywhere, not just here.

Comment (worldofpeace) [reply pkgs/development/python-modules/digitalocean/default.nix]: @jonringer Would you care to open an issue referencing our discussion?

**Comment by worldofpeace** [comment]
Thank you @Kiwi

**Pull request merged**
